cognito-post-authentication/src/test/java/no/unit/nva/cognito/PostAuthenticationHandlerTest.java

```diff
@@ -47,80 +48,125 @@
     @BeforeEach
     public void init() {
         customerApi = mock(CustomerApi.class);
-        userApi = mock(UserApi.class);
+        userApi = new UserApiMock();
         awsCognitoIdentityProvider = mock(AWSCognitoIdentityProvider.class);
         userService = new UserService(userApi, awsCognitoIdentityProvider);
         handler = new PostAuthenticationHandler(userService, customerApi);
     }
 
     @Test
-    public void handleRequestUpdatesUserPoolWithExistingUserWhenUserIsFound() {
-        UUID customerId = UUID.randomUUID();
-        prepareMocksWithExistingCustomer(customerId);
-        prepareMocksWithExistingUser(SAMPLE_FEIDE_ID);
+    public void handleRequestUsesExistingUserWhenUserIsFound() {
+        prepareMocksWithExistingCustomer();
+        prepareMocksWithExistingUser();
 
         Event requestEvent = createRequestEvent();
         final Event responseEvent = handler.handleRequest(requestEvent, mock(Context.class));
 
-        verify(userApi, times(0)).createUser(any());
-        verify(awsCognitoIdentityProvider).adminAddUserToGroup(any());
+        verify(awsCognitoIdentityProvider, times(2)).adminAddUserToGroup(any());
         verify(awsCognitoIdentityProvider).adminUpdateUserAttributes(any());
 
+        assertEquals(getUserFromMock(), createUserWithInstitutionAndCreatorRole());
```

Comment:
assign the expectedValue to a variable named `expected` and getUserFromMock to a vaiable named `createdUser`

Since you use this line a lot you can  first refactor the line to own method and then refactor the method adding the aforementioned variables